PR: https://github.com/odoo/odoo/pull/

From: 2c29168307c433a114af7c3a947615bd507a8631
From: Ravi Gohil
Date: 2016-06-30 12:41:21

Structural Changes: 1
Total Changes: 39

[MOV] project: split and restructured files according to new API guidelines

- created report/project_cumulative.py for model
  project.task.history.cumulative that was in project.py
- moved project.project kanban view from project_dashboard.xml
  to project_views.xml, which required moving some menuitems
  as well so they are available while refering them in other files

================================= pseudo patch: =================================

--- a/addons/project/models/project.py
+++ b/addons/project/models/project.py
@@ -913,45 +913,6 @@ class project_task_history(osv.osv):
         'date': fields.date.context_today,
     }
 
-class project_task_history_cumulative(osv.osv):
-    _name = 'project.task.history.cumulative'
-    _table = 'project_task_history_cumulative'
-    _inherit = 'project.task.history'
-    _auto = False
-
-    _columns = {
-        'end_date': fields.date('End Date'),
-        'nbr_tasks': fields.integer('# of Tasks', readonly=True),
-        'project_id': fields.many2one('project.project', 'Project'),
-    }
-
-    def init(self, cr):
-        tools.drop_view_if_exists(cr, 'project_task_history_cumulative')
-
-        cr.execute(""" CREATE VIEW project_task_history_cumulative AS (
-            SELECT
-                history.date::varchar||'-'||history.history_id::varchar AS id,
-                history.date AS end_date,
-                *
-            FROM (
-                SELECT
-                    h.id AS history_id,
-                    h.date+generate_series(0, CAST((coalesce(h.end_date, DATE 'tomorrow')::date - h.date) AS integer)-1) AS date,
-                    h.task_id, h.type_id, h.user_id, h.kanban_state,
-                    count(h.task_id) as nbr_tasks,
-                    greatest(h.remaining_hours, 1) AS remaining_hours, greatest(h.planned_hours, 1) AS planned_hours,
-                    t.project_id
-                FROM
-                    project_task_history AS h
-                    JOIN project_task AS t ON (h.task_id = t.id)
-                GROUP BY
-                  h.id,
-                  h.task_id,
-                  t.project_id
-
-            ) AS history
-        )
-        """)
 
 class project_tags(osv.Model):
     """ Tags of project's tasks (or issues) """
